### PR TITLE
BACKLOG-13201 add permission for advanced options tab

### PR DESCRIPTION
**Pull request by Taybou**
## JIRA

https://jira.jahia.org/browse/BACKLOG-13201

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/EditPanel/EditPanel.jsx [hunk 1]
```diff
@@ -70,13 +82,15 @@ const EditPanelCmp = ({formik, title, notificationContext, client}) => {
...
     const SelectedTabComponents = {};
-    tabActions.forEach(action => {
-        SelectedTabComponents[action.value] = action.displayableComponent;
-    });
-
-    const [activeTab, setActiveTab] = useState('edit');
+    const tabActions = registry.find({target: 'editHeaderTabsActions'});
+    tabActions
+        .filter(tab => tab.value !== 'advanced' ||
+            (tab.value === 'advanced' && advancedTabPermission.checksResult)
```
Comment (dgriffon) [inline src/javascript/EditPanel/EditPanel.jsx:89]: As tabs are now components in the registry, the permission must be set on the tab registration (in the config) and the check must be done on each tab.

File: src/javascript/EditPanel/EditPanel.jsx [hunk 2]
```diff
@@ -9,15 +9,23 @@ import {HeaderLowerSection, HeaderUpperSection} from './header';
...
     const {t} = useTranslation();
-    const {nodeData, siteInfo, lang, uilang, mode, nodeTypeName, refetchFormData} = useContentEditorContext();
+    const {nodeData, siteInfo, site, lang, uilang, mode, nodeTypeName, refetchFormData} = useContentEditorContext();
     const {envProps} = useContentEditorConfigContext();
+    const [activeTab, setActiveTab] = useState('edit');
+
+    const advancedTabPermission = useNodeChecks(
+        {path: `/sites/${site}`},
+        {requiredPermission: [Constants.permissions.canSeeAdvancedOptionsTab]}
+    );
```
Comment (dgriffon) [inline src/javascript/EditPanel/EditPanel.jsx:28]: This has to be set at tab registration level, not hardcoded here.

Comment (Taybou) [reply src/javascript/EditPanel/EditPanel.jsx:28]: is there any doc how to use permissions with registry?

Comment (Taybou) [reply src/javascript/EditPanel/EditPanel.jsx:28]: or any similar working code?

Comment (dgriffon) [reply src/javascript/EditPanel/EditPanel.jsx:28]: yes, you can find an example of an action with restriction:
https://github.com/Jahia/content-editor/blob/master/src/javascript/Create/CreateNewContentAction/createNewContent.action.jsx#L48

Comment (dgriffon) [reply src/javascript/EditPanel/EditPanel.jsx:28]: The check has to be done in the advance option component

Comment (Taybou) [reply src/javascript/EditPanel/EditPanel.jsx:28]: I'll see

**Pull request merged**